psamm/randomsparse.py

```diff
@@ -0,0 +1,207 @@
+        self._model = model
+        if reaction_set is None:
+            self._test_set = set(self._model.reactions)
+        else:
+            self._test_set = set(reaction_set)
+        self._reactions = set(self._test_set)
+
+    def get_total(self):
+        return self._reactions
+
+    def tests(self):
+        while len(self._test_set) > 0:
+            reaction = random.choice(tuple(self._test_set))
+            self._test_set.remove(reaction)
+            yield reaction, {reaction}
+
+    def delete(self, entity, deleted_reactions):
+        pass
+
+
+class GeneDeletionStrategy(object):
+    """Deleting genes strategy class.
+    """
+
+    def __init__(self, model, gene_assoc):
+        self._model = model
+        self._genes = set()
+        self._gene_assoc = dict(gene_assoc)
+
+        for reaction, assoc in iteritems(self._gene_assoc):
+            self._genes.update(v.symbol for v in assoc.variables)
+
+        self._reactions = set(self._model.reactions)
+        self._test_set = set(self._genes)
+
+    def get_total(self):
+        return set(self._genes)
+
+    def tests(self):
+        while len(self._test_set) > 0:
+            gene = random.choice(tuple(self._test_set))
+            self._test_set.remove(gene)
+
+            deleted_reactions = set()
+            self._new_gene_assoc = {}
+            for reaction in self._reactions:
+                if reaction not in self._gene_assoc:
+                    continue
+                assoc = self._gene_assoc[reaction]
+                if boolean.Variable(gene) in assoc.variables:
+                    new_assoc = assoc.substitute(
+                        lambda v: v if v.symbol != gene else False)
+                    if new_assoc.has_value() and not new_assoc.value:
+                        deleted_reactions.add(reaction)
+                    else:
+                        self._new_gene_assoc[reaction] = new_assoc
+                else:
+                    self._new_gene_assoc[reaction] = assoc
+
+            yield gene, deleted_reactions
+
+    def delete(self, entity, deleted_reactions):
+        self._reactions -= deleted_reactions
+        self._gene_assoc = self._new_gene_assoc
+
+
+def get_gene_associations(model):
+    """Create gene association for class :class:`.GeneDeletionStrategy`.
+
+    Args:
+        model: :class:`psamm.datasource.native.NativeModel`.
+    """
+
+    for reaction in model.parse_reactions():
+        assoc = None
+        if reaction.genes is None:
+            continue
+        elif isinstance(reaction.genes, string_types):
+            assoc = boolean.Expression(reaction.genes)
+        else:
+            variables = [boolean.Variable(g) for g in reaction.genes]
+            assoc = boolean.Expression(boolean.And(*variables))
+        yield reaction.id, assoc
+
+
+def get_exchange_reactions(model):
+    """Create reaction ID set for :class:`.GeneDeletionStrategy` if
+    'exchange' is set up as the deletion type.
```

Comment:
Describe what the function does in the summary, e.g. "Yield all reaction IDs from model that are exchange reactions". In the next paragraph you can describe that the function is useful for the `GeneDeletionStrategy`. I would also add a note in the `GeneDeletionStrategy` saying that this function can be useful.